homodeus_motivations/scripts/motv_speech_recognition_observer.py

```diff
@@ -30,57 +47,112 @@ def listen_desires_set_Cb(self,desireSet):
         """
         # S'il y a beaucoup de desirs du meme type... il me semble que ca ne fonctionnera pas
         event = Event()
+        rospy.loginfo(self.curDesireSet.desires)
         for desire in self.curDesireSet.desires:
-            if desire.type == "Speech_recognition":
+            if desire.type == "Listening":
                 event.desire = desire.id
                 event.desire_type = desire.type
-                if self.accomplish_criterion(SpeechText.data):
-                    event.type = Event.ACC_ON
-                    self.event_publisher.publish(event)
-                elif self.cancel_criterion():
-                    #Probleme actuel est que si le desir n'est pas active il peut tout de meme changer d'etat!
-                    event.type = Event.IMP_ON
-                    self.event_publisher.publish(event)
+                event.type = Event_type
+                self.event_publisher.publish(event)
+
+        #reset 
+        self.timer.cancel()
+        self.actual_context=""
                 
+    def listen_Cb(self,Stt_text):
+        """
+        This method is the Cb of the speech_recognition perception
+        Arguments
+        ---------
+        Stt_text: String()
+            the string received from speech_recognition perception
+        """
 
+        text_to_use = no_caps_and_whitespace(Stt_text.data)
+        if self.actual_context:
+            if self.actual_context !="menu":
+                self._look_answer(text_to_use)
+            else:
+                self._look_answer_menu(text_to_use)
+        else:
+            rospy.loginfo("Listening without receiving a context")
 
-    def accomplish_criterion(self, SpeechText):
+    def _look_answer(self,Stt_text):
         """
-        This method look the result of the module and return True if it accomplished the criterion the motivation is looking for
-
+        This method analyse if the desire is accomplish or not depending on the context and the xml doc
         Arguments
         ---------
-        SpeechText : string
-            The text recognizes by the robot, it will be analyzed to know if something 
-            relevant have been said and considerer if the desire is now accomplished
+        Stt_text: string
+            the string received from speech_recognition perception
         """
-        # Pourrait etre un xml avec les mots a retrouves selon le contexte et le desir donne le contexte
-        if 'scenario' in str(SpeechText):
-            return True
-        else:
-            return False
+        xml_path_to_search= "./"+self.actual_context+"/answer"
+        for answer in self.speech_context_parsed.findall(xml_path_to_search):
+            if re.search(r"\b{}\b".format(answer.text), Stt_text) is not None:
+                if answer.get('acc_type') == 'ON':
+                    self.desire_event_change(Event.ACC_ON)
+                    return
```

Comment:
in certain programming philosophies functions have one return statement (in most cases), if the reason is only to leave the loop once the answer is found without searching for all the remaining words, the break statement could be considered